commit 909a40b63b13bf2fb0e626740428828ac9a01986
Author: Rainer Jung <rjung@apache.org>
Date:   Fri Sep 11 20:51:00 2009 +0000

    Switch to volatile for status flag used by multiple
    threads during cluster context startup.
    
    Without volatile problems have been observed, where the
    change by one thread was actually not seen by the other
    thread.
    
    Performance implications are neglectable, because the
    variable is only used during context startup.
    
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@814024 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/catalina/ha/session/DeltaManager.java b/java/org/apache/catalina/ha/session/DeltaManager.java
index 8bf7b98..3eeab77 100644
--- a/java/org/apache/catalina/ha/session/DeltaManager.java
+++ b/java/org/apache/catalina/ha/session/DeltaManager.java
@@ -112,7 +112,7 @@ public class DeltaManager extends ClusterManagerBase{
     private boolean expireSessionsOnShutdown = false;
     private boolean notifyListenersOnReplication = true;
     private boolean notifySessionListenersOnReplication = true;
-    private boolean stateTransfered = false ;
+    private volatile boolean stateTransfered = false ;
     private int stateTransferTimeout = 60;
     private boolean sendAllSessions = true;
     private boolean sendClusterDomainOnly = true ;
